src/aioftp/pathio.py

```diff
@@ -1,21 +1,43 @@
 import abc
 import asyncio
-import collections
 import functools
 import io
 import operator
-import pathlib
 import stat
 import sys
 import time
+from collections.abc import AsyncIterable, Awaitable, Callable, Generator, Iterator
+from concurrent.futures import Executor
+from os import stat_result
+from pathlib import Path, PurePath, PurePosixPath
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Generic,
+    Literal,
+    NamedTuple,
+    Protocol,
+    TypedDict,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from typing_extensions import Concatenate, ParamSpec, Self, TypeAlias, Unpack
 
-from . import errors
 from .common import (
     DEFAULT_BLOCK_SIZE,
     AbstractAsyncLister,
     AsyncStreamIterator,
     with_timeout,
 )
+from .errors import PathIOError
+
+if TYPE_CHECKING:
+    from _typeshed import OpenBinaryMode, ReadableBuffer
+
+    from .server import Connection
```

Comment:
If this `if TYPE_CHECKING` is about circular imports, than lets resolve this via moving Connection to common module